docs/code/IDataViewTypeSystem.md

```diff
@@ -0,0 +1,844 @@
+Columns that are not involved in the transformation are simply "passed
+through" from the source `IDataView` to the new one.
+
+Detailed specifications of the `IDataView`, `ISchema`, and `IRowCursor`
+interfaces are in other documents.
+
+### Column Types
+
+Each column in an `IDataView` has an associated column type. The collection of
+column types is open, in the sense that new code can introduce new column
+types without requiring modification of all `IDataView` related components.
+While introducing new types is possible, we expect it will also be relatively
+rare.
+
+All column type implementations derive from the abstract class `ColumnType`.
+Primitive column types are those whose implementation derives from the
+abstract class `PrimitiveType`, which derives from `ColumnType`.
+
+### Representation Type
+
+A column type has an associated .Net type, known as its representation type or
+raw type.
+
+Note that a column type often contains much more information than the
+associated .Net representation type. Moreover, many distinct column types can
+use the same representation type. Consequently, code should not assume that a
+particular .Net type implies a particular column type.
+
+### Standard Column Types
+
+There is a set of predefined standard column types, divided into standard
+primitive types and vector types. Note that there can be types that are
+neither primitive nor vector types. These types are not standard types and may
+require extra care when handling them. For example, a `PictureType` value
+might require disposing when it is no longer needed.
+
+Standard primitive types include the text type, the boolean type, numeric
+types, and key types. Numeric types are further split into floating-point
+types, signed integer types, and unsigned integer types.
+
+A vector type has an associated item type that must be a primitive type, but
+need not be a standard primitive type. Note that vector types are not
+primitive types, so vectors of vectors are not supported. Note also that
+vectors are homogeneous—all elements are of the same type. In addition to its
+item type, a vector type contains dimensionality information. At the basic
+level, this dimensionality information indicates the length of the vector
+type. A length of zero means that the vector type is variable length, that is,
+different values may have different lengths. Additional detail of vector types
+is in a subsequent section. Vector types are instances of the sealed class
+`VectorType`, which derives from `ColumnType`.
+
+This document uses convenient shorthand for standard types:
+
+* `TX`: text
+
+* `BL`: boolean
+
+* `R4`, `R8`: single and double precision floating-point
+
+* `I1`, `I2`, `I4`, `I8`: signed integer types with the indicated number of
+bytes
+
+* `U1`, `U2`, `U4`, `U8`: unsigned integer types with the indicated number of
+bytes
+
+* `UG`: unsigned type with 16-bytes, typically used as an unique ID
```

Comment:
"as an unique" should be "as a unique"

... yes I had to look this one up to make sure I'm not crazy: https://www.quora.com/Which-is-the-correct-grammar-usage-a-unique-or-an-unique/answer/Stephanie-Fysh?share=1